### ISSUE TITLE
[BLAZOR] WASM 404 when deployed to IIS

### ISSUE THREAD

**Issue by eli191** [dotnet/aspnetcore#41028]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Blazor was app give the following errors when deployed to IIS. The IIS contains the .Net hosting bundle. Works fine locally.

It is a brand new app, just created with Visual Studio and deployed to IIS, no code added.

<img width="547" alt="Screenshot 2022-04-04 at 15 10 42" src="https://user-images.githubusercontent.com/16271512/161551067-dbd968b3-a1b5-4cc6-9b53-7b040bbe897a.png">

I am using 
Visual Studio Community 2022 for Mac Preview
Version 17.0 Preview (17.0 build 8320)
Runtime .NET 6.0.3 (64-bit)

Please help me find a solution for this.

Thanks!

### Expected Behavior

The app is supposed to work well on the server, as it does when launched in debug.

### .NET Version

6.0.201

**Comment by javiercn** [comment]
@eli191 thanks for contacting us.

* Is this a standalone project or a hosted project? 
* Are the files present on the wwwroot folder of the application?
* On the network tab in the browser developer tools, are there 404 responses?

**Comment by eli191** [comment]
@javiercn 

> * Is this a standalone project or a hosted project?
Standalone
> * Are the files present on the wwwroot folder of the application?
Yes
> * On the network tab in the browser developer tools, are there 404 responses?
Only those 2 files ending with .dat and .blat respond 404, but they are present on the server inside the _framework folder.

**Comment by eli191** [comment]
WORKING NOW!!

I have added the [URL Rewrite Module 2.1](https://www.iis.net/downloads/microsoft/url-rewrite) as suggested by @Sharaf-Mansour .
And I have added MIME mappings in IIS for the extensions .dat and .blat with the value application/octet-stream.
The answer by @javiercn put me on the right track because in the network tab in the browser developer tools, it was written that maybe a MIME mapping was missing, and indeed, those 2 mappings were missing.

Thanks everyone!
